prelude/bootstrap.js

```diff
@@ -585,13 +632,37 @@ function payloadFileSync(pointer) {
     }
   }
 
+  let bypassCompressCheckWhenCallbyCreateReadStream = false;
```

Comment:
Is it possible to have a race condition with this "global" variable? 

What will happen if multiple threads call related functions, and some of them call `createReadStream` while others don't?

Reply:
I though of this as well. So far, my investigation shows that;
 - this javascript code is only used by the nodejs thread, meaning that there will be only one call of  createReadStream at a time.
 -  the fs.createReadStream  is calling fs.open without async nor callback call in between. this means that the function execution is atomic and that a call to fs.createReadStream leads to a call to fs.open without going back to the eventloop.
 
- obviously, this assumption could be defeated if the nodejs guys decide to change their implementations.

Reply:
This may happen when using worker threads or clusters I think?

Reply:
I think that each worker thread will have their own copy of the global variable.